### PR TITLE
Load room history if necessary when searching for MSC3089 getFileEvent()

### PR DESCRIPTION
**Pull request by hughns**
---
Here's what your changelog entry will look like:

## ✨ Features
 * Load room history if necessary when searching for MSC3089 getFileEvent() ([\#2066](https://github.com/matrix-org/matrix-js-sdk/pull/2066)). Contributed by @hughns.

### CODE REVIEW

File: src/models/MSC3089Branch.ts [hunk 1]
```diff
@@ -140,7 +140,14 @@ export class MSC3089Branch {
         const room = this.client.getRoom(this.roomId);
         if (!room) throw new Error("Unknown room");
 
-        const event = room.getUnfilteredTimelineSet().findEventById(this.id);
+        let event: MatrixEvent | undefined = room.getUnfilteredTimelineSet().findEventById(this.id);
+
+        // keep scrolling back if needed until we find the event or reach the start of the room:
+        while (!event && room.oldState.paginationToken) {
```
Comment (turt2live) [inline src/models/MSC3089Branch.ts:146]: do we specifically want `oldState`, or `currentState`? Using `oldState` is sufficiently rare where it should get a comment when we deliberately use it.

**Review by turt2live — CHANGES_REQUESTED** [review]
otherwise lgtm

File: src/models/MSC3089Branch.ts (hunk 1 above)
Comment (hughns) [reply src/models/MSC3089Branch.ts]: Good spot. `currentState` makes more sense and I've also used the non-legacy implementation.

**Review by turt2live — APPROVED** [review]
Thanks :D

**Pull request merged**